### PR TITLE
[MIR trans] Add support for SwitchInt

### PR DESCRIPTION
**Pull request by dotdash**

### CODE REVIEW

File: src/librustc_trans/trans/mir/block.rs [hunk 1]
```diff
@@ -50,8 +50,13 @@ impl<'bcx, 'tcx> MirContext<'bcx, 'tcx> {
...
             }
 
-            mir::Terminator::SwitchInt { .. } => {
-                unimplemented!()
+            mir::Terminator::SwitchInt { ref discr, switch_ty, ref values, ref targets } => {
+                let (otherwise, targets) = targets.split_last().unwrap();
+                let discr = build::Load(bcx ,self.trans_lvalue(bcx, discr).llval);
+                let switch = build::Switch(bcx, discr, self.llblock(*otherwise), values.len());
+                for (value, target) in values.iter().zip(targets) {
+                    build::AddCase(switch, self.trans_constval(bcx, value, switch_ty), self.llblock(*target))
```
Comment (nagisa) [inline src/librustc_trans/trans/mir/block.rs]: Overlong line?

Comment (dotdash) [inline src/librustc_trans/trans/mir/block.rs]: Indeed, fixed, thanks!

File: src/librustc_trans/trans/mir/block.rs [hunk 2]
```diff
@@ -50,8 +50,15 @@ impl<'bcx, 'tcx> MirContext<'bcx, 'tcx> {
                 unimplemented!()
             }
 
-            mir::Terminator::SwitchInt { .. } => {
-                unimplemented!()
+            mir::Terminator::SwitchInt { ref discr, switch_ty, ref values, ref targets } => {
+                let (otherwise, targets) = targets.split_last().unwrap();
+                let discr = build::Load(bcx ,self.trans_lvalue(bcx, discr).llval);
```
Comment (nikomatsakis) [inline src/librustc_trans/trans/mir/block.rs]: Nit: funny comma here 

`s/bcx ,/bcx, /`

**Comment by nikomatsakis** [comment]
r=me when the nit is fixed, nice!

**Comment by dotdash** [comment]
@bors retry

2015-11-10 12:34 GMT+01:00 bors <notifications@github.com>:

**Pull request merged**